test/genlib_retry_tests.erl

```diff
@@ -10,6 +10,14 @@
 
 %%
 
+-spec new_test() -> _.
+new_test() ->
+    ?assertEqual(genlib_retry:linear(3, 64), genlib_retry:new({linear, 3, 64})),
+    ?assertEqual(genlib_retry:exponential(3, 2, 64), genlib_retry:new({exponential, 3, 2, 64})),
+    ?assertEqual(genlib_retry:exponential(3, 2, 64, 500), genlib_retry:new({exponential, 3, 2, 64, 500})),
+    ?assertEqual(genlib_retry:intervals([3]), genlib_retry:new({intervals, [3]})),
+    ?assertEqual(genlib_retry:timecap(10, {linear, 3, 64}), genlib_retry:timecap(10, genlib_retry:linear(3, 64))).
```

Comment:
Is it me or is it rather not the best idea to compare opaque terms? The `timecap` strategy for example stores current timestamp here (well, maybe not the best idea either). Should we just deprecate legacy constructors in favor of `new/1` and not test them at all?